nixos/modules/services/web-apps/convos.nix

```
@@ -0,0 +1,80 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.convos;
+in
+{
+  options.services.convos = {
+    enable = mkEnableOption "Convos";
+    package = mkOption {
+      default = pkgs.convos;
+      defaultText = "pkgs.convos";
+      type = types.package;
+      description = "Convos package to use";
+    };
+    listenPort = mkOption {
+      type = types.int;
+      default = 3000;
+      example = 8080;
+      description = "Port the web interface should listen on";
+    };
+    listenAddress = mkOption {
+      type = types.str;
+      default = "*";
+      example = "127.0.0.1";
+      description = "Address or host the web interface should listen on";
+    };
+    reverseProxy = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enables reverse proxy support. This will allow Convos to automatically
+        pick up the X-Forwarded-For and X-Request-Base HTTP headers set in your
+        reverse proxy web server. Note that setting this environment variable
+        without a reverse proxy in front will be a security issue.
+      '';
+    };
+  };
+
+
+  config = mkIf cfg.enable {
+    systemd.services.convos = {
+      description = "Convos Service";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "networking.target" ];
+      serviceConfig = {
+        User = "convos";
+        Group = "convos";
+        Environment = [
+          "CONVOS_HOME=%S/convos"
+          "CONVOS_REVERSE_PROXY=${if cfg.reverseProxy then "1" else "0"}"
+          "MOJO_LISTEN=http://${toString cfg.listenAddress}:${toString cfg.listenPort}"
+        ];
```

You could move this chunk of configuration from `serviceConfig` into `systemd.services.convos` and write it the following way instead:

```suggestion
        environment = {
          CONVOS_HOME = "%S/convos";
          CONVOS_REVERSE_PROXY = if cfg.reverseProxy then "1" else "0"; # I'm guessing simply putting cfg.reverseProxy; would even work here...
          MOJO_LISTEN = "http://${toString cfg.listenAddress}:${toString cfg.listenPort}";
        };
```

---

Thanks, that seems a bit nicer! Will do